components/configManager.ts

```diff
@@ -0,0 +1,452 @@
+        return configManager._require("static/Scpcallunlockables.json")
+    },
+    ScpcLocationsData() {
+        return configManager._require("static/ScpcLocationsData.json")
+    },
+    DiscordRichAssetsForBricks() {
+        return configManager._require("static/DiscordRichAssetsForBricks.json")
+    },
+    EscalationCodenames() {
+        return configManager._require("static/EscalationCodenames.json")
+    },
+    ScoreOverviewTemplate() {
+        return configManager._require("static/ScoreOverviewTemplate.json")
+    },
+    PeacockGameChangerProperties() {
+        return configManager._require(
+            "static/PeacockGameChangerProperties.json",
+        )
+    },
+    MultiplayerPresets() {
+        return configManager._require("static/MultiplayerPresets.json")
+    },
+    LobbySlimTemplate() {
+        return configManager._require("static/LobbySlimTemplate.json")
+    },
+    MasteryDataForLocationTemplate() {
+        return configManager._require(
+            "static/MasteryDataForLocationTemplate.json",
+        )
+    },
+    LegacyMasteryLocationTemplate() {
+        return configManager._require(
+            "static/LegacyMasteryLocationTemplate.json",
+        )
+    },
+    DefaultCpdConfig() {
+        return configManager._require("static/DefaultCpdConfig.json")
+    },
+    EvergreenGameChangerProperties() {
+        return configManager._require(
+            "static/EvergreenGameChangerProperties.json",
+        )
+    },
+    AreaMap() {
+        return configManager._require("static/AreaMap.json")
+    },
+    ArcadePageTemplate() {
+        return configManager._require("static/ArcadePageTemplate.json")
+    },
+    HitsCategoryElusiveTemplate() {
+        return configManager._require("static/HitsCategoryElusiveTemplate.json")
+    },
+    HitsCategoryContractAttackTemplate() {
+        return configManager._require(
+            "static/HitsCategoryContractAttackTemplate.json",
+        )
+    },
+    MissionRewardsTemplate() {
+        return configManager._require("static/MissionRewardsTemplate.json")
+    },
+    SniperUnlockables() {
+        return configManager._require("static/SniperUnlockables.json")
+    },
+}
+
+export type ConfigKey = keyof typeof configs
+
+export const configKeys: ConfigKey[] = Object.keys(configs) as ConfigKey[]
+
+/**
+ * Get a config file.
+ * Configs for H1 start with "Legacy", "H2" for HITMAN 2, and no prefix for HITMAN 3.
+ *
+ * @param config The name of the config file.
+ * @param clone If the value should be cloned (saves memory if false, but as a side effect, modifications will affect the actual config).
+ * @returns The config.
+ * @throws {Error} If the config file specified doesn't exist.
+ */
+export function getConfig<T = unknown>(config: ConfigKey, clone: boolean): T {
+    if (!Object.hasOwn(configs, config)) {
+        throw new Error(`Tried to lookup config that does not exist: ${config}`)
+    }
+
+    let c = configs[config]()
```

Comment:
So, we're fine this will modify the original object whenever clone is set to false? And if so, what will happen if a plugin gets a config that it already modified before? I'm aware this is how it worked originally, but we didn't have plugins modifying the false (yet), then, only core. And we are/were very careful to make sure we clone when we do that... Seems like a can of worms.

Reply:
This will modify the original object even if clone is true. It's a matter of do we want to always clone (very slow for big objects like the H1 templates) or only clone when we know the consumer code is going to modify the object being returned.

Reply:
I think in the end it's not really our responsibility. I do think that in the case of `clone == true`, we should tap the cloned object, not the source object. Since we always pass `clone` to the tap, it's up to the plugin to handle it gracefully.

I'm trying to think of situations where one might want to apply different logic based on `clone`, but can't really think of one right now. Maybe a non-issue?

Reply:
So I've been thinking on this, maybe we just have configs be modified by plugins when they are first (lazily) loaded? I can't think of any plugins that need to modify a config, say, after a specific server event is fired.